src/telemetry/unhandledErrorReporter.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { extractErrorDetails, RawTelemetryReporter } from './telemetryReporter';
+
+type ExceptionType = 'uncaughtException' | 'unhandledRejection';
+
+export function installUnhandledErrorReporter(telemetryReporter: RawTelemetryReporter): void {
+  process.addListener('uncaughtException', (exception: unknown) => {
```

Comment:
I believe this is run in the extension host within vscode, so we may end up getting exceptions thrown by other exceptions here. We probably want to filter these out, perhaps by checking its stacktrace to ensure it's coming from one of our scripts?

Reply:
I am also still not sure whether we should be moving it out of the EH to run in a separate process like other debug adapters.

Reply:
Done